payroll/models/hr_payslip.py

```diff
@@ -711,55 +747,73 @@ def get_payslip_vals(
         )
         return res
 
-    @api.onchange("employee_id", "date_from", "date_to", "struct_id")
-    def onchange_employee(self):
+    def _get_employee_contracts(self):
+        return self.env["hr.contract"].browse(
+            self.employee_id._get_contracts(
+                date_from=self.date_from, date_to=self.date_to
+            ).ids
+        )
 
-        if (not self.employee_id) or (not self.date_from) or (not self.date_to):
+    @api.onchange("struct_id")
+    def onchange_struct_id(self):
+        if not self.struct_id:
             return
+        input_lines = self.input_line_ids.browse([])
+        input_line_ids = self.get_inputs(
+            self._get_employee_contracts(), self.date_from, self.date_to
+        )
+        for r in input_line_ids:
+            input_lines += input_lines.new(r)
+        self.input_line_ids = input_lines
 
-        employee = self.employee_id
-        date_from = self.date_from
-        date_to = self.date_to
-        contract_ids = self.contract_id.ids
-
-        ttyme = datetime.combine(date_from, time.min)
-        locale = self.env.context.get("lang") or "en_US"
-        self.name = _("Salary Slip of %s for %s") % (
-            employee.name,
-            tools.ustr(
-                babel.dates.format_date(date=ttyme, format="MMMM-y", locale=locale)
-            ),
+    @api.onchange("date_from", "date_to")
+    def onchange_dates(self):
+        if not self.date_from or not self.date_to:
+            return
```

Comment:
Same comment, if the dates are false it will be defaulted to current month period when you select the employee. 

Note that this function only runs with onchange and when you first select the employee the date is automatically default to current month. If the user change the date manually, we then don't want to change it again when he changes other field. 

The purpose of that returns is to preserve user selection. Before this change if you change struct_id or dates everything is defaulted again which is inconvenient for the user.

Reply:
Understood.